src/battle_regulation.c

```diff
@@ -0,0 +1,276 @@
+    }
+
+    rule = LinkBattleRuleset_GetRuleValue(ruleset, 13);
+    if (rule == 0x8000) {
+        return SpeciesIsMythical(species) != TRUE;
+    }
+
+    rule = LinkBattleRuleset_GetRuleValue(ruleset, 11);
+    if (rule == 0) {
+        if (MonCheckFrontierIneligibility(pokemon)) {
+            return FALSE;
+        }
+    }
+
+    return TRUE;
+}
+
+int sub_02074A6C(LinkBattleRuleset *a0, Party *party, PokedexData *pokedex, u8 *selectedOrder) {
+    Pokemon *mon;
+    int rule;
+    int numMons = 0;
+    int j;
+    int i;
+    int totalLevel = 0;
+    u16 species[PARTY_SIZE];
+    u16 heldItems[PARTY_SIZE];
+
+    if (a0 == NULL) {
+        return FALSE;
+    }
+
+    for (i = 0; i < PARTY_SIZE; ++i) {
+        species[i]   = SPECIES_NONE;
+        heldItems[i] = ITEM_NONE;
+        if (selectedOrder[i] != 0) {
+            ++numMons;
+        }
+    }
+    rule = LinkBattleRuleset_GetRuleValue(a0, 1);
+    if (numMons != rule) {
+        return 4;
+    }
+
+    for (i = 0; i < PARTY_SIZE; ++i) {
+        if (selectedOrder[i] != 0) {
+            mon = Party_GetMonByIndex(party, selectedOrder[i] - 1);
+            if (!sub_0207496C(a0, mon, pokedex)) {
+                return 5;
+            }
+            species[i]   = GetMonData(mon, MON_DATA_SPECIES, NULL);
+            heldItems[i] = GetMonData(mon, MON_DATA_HELD_ITEM, NULL);
+            totalLevel += GetMonData(mon, MON_DATA_LEVEL, NULL);
+        }
+    }
+    rule = LinkBattleRuleset_GetRuleValue(a0, 3);
+    if (totalLevel > rule && rule != 0) {
+        return 1;
+    }
+
+    rule = LinkBattleRuleset_GetRuleValue(a0, 13);
+    if (rule == 0x8000) {
+        j = 0;
+        for (i = 0; i < PARTY_SIZE; ++i) {
+            if (species[i] != SPECIES_NONE && IsPokemonBannedFromBattleFrontier(species[i], 0) && !SpeciesIsMythical(species[i])) {
+                ++j;
+            }
+        }
+        if (j > 2) {
+            return 6;
+        }
+    }
+
+    rule = LinkBattleRuleset_GetRuleValue(a0, 10);
+    if (rule == 0 && numMons > 1) {
+        for (i = 0; i < PARTY_SIZE - 1; ++i) {
+            for (j = i + 1; j < PARTY_SIZE; ++j) {
+                if (species[i] == species[j] && species[i] != SPECIES_NONE) {
+                    return 2;
+                }
+            }
+        }
+    }
+
+    rule = LinkBattleRuleset_GetRuleValue(a0, 13);
+    if (rule == 0x8000) {
+        for (i = 0; i < PARTY_SIZE; ++i) {
+            if (heldItems[i] == ITEM_SOUL_DEW && species[i] != SPECIES_NONE && heldItems[i] != ITEM_NONE) {
+                return 8;
+            }
+        }
+    }
+
+    rule = LinkBattleRuleset_GetRuleValue(a0, 9);
+    if (rule == 0 && numMons > 1) {
+        for (i = 0; i < PARTY_SIZE - 1; ++i) {
+            for (j = i + 1; j < PARTY_SIZE; ++j) {
+                if (heldItems[i] == heldItems[j] && species[i] != SPECIES_NONE && heldItems[i] != ITEM_NONE) {
+                    return 3;
+                }
+            }
+        }
+    }
+
+    return 0;
```

Comment:
RULESET_ERROR_NONE
^error consts like this perhaps

Reply:
Will handle in #332

Reply:
jk will handle now :)